### PR TITLE
Hide community channels when hide channels option is enabled

### LINKED ISSUE #19377: Hidden channel is displayed for users who do not have permission to access it
Recently Desktop has implemented a new feature - 'Hiding channels from members who do not have permissions to view them' 
Status Desktop 2024-03-22 20-26-36

Looks like this feature has not been implemented yet on mobile, because hidden channels are still displayed in channel list for users who do not have permissions to view them. 

**Steps:** 

1. On Desktop create token gated channel and enable 'Hide channel' option
2. Share community to mobile user who does not have required tokens to view channel
3. Join community by mobile user 
4. See if hidden channel is displayed in the list of channels

**Expected result:** hidden channel should not be displayed

**Actual result:** hidden channel is displayed 

[Desktop_logs.log.zip](https://github.com/status-im/status-mobile/files/14727155/Desktop_logs.log.zip)
[mobile_logs.zip](https://github.com/status-im/status-mobile/files/14727156/mobile_logs.zip)
photo_2024-03-22 20 32 37
Status Desktop 2024-03-22 20-32-19

### Additional Information

- Status version: nightly 
- Operating System: Android, iOS

[comment]: # (Please, add logs/notes if necessary)

### PR DESCRIPTION
**Pull request by ibrkhalil**
fixes #19377 

### Summary

Hide channels when "hideIfPermissionsNotMet" flag is enabled and token-gating permissions are not fullfiled.

status: ready

### CODE REVIEW

**Review by seanstrom — APPROVED** [review]
Looks good! 🙌

File: src/status_im/subs/communities.cljs [hunk 1]
```diff
@@ -243,7 +243,8 @@
                 emoji
                 can-view?
                 can-post?
-                token-gated?]}]]
+                token-gated?
+                hideIfPermissionsNotMet]}]]
```
Comment (flexsurfer) [inline src/status_im/subs/communities.cljs]: `:hide-if-permissions-not-met?`

File: src/status_im/subs/communities.cljs [hunk 2]
```diff
@@ -259,19 +260,21 @@
...
+                             :mentions-count               (or unviewed-mentions-count 0)
+                             :can-post?                    can-post?
                              ;; NOTE: this is a troolean nil->no permissions, true->no access, false
                              ;; -> has access
-                             :locked?          (when token-gated?
-                                                 (and (not can-view?)
-                                                      (not can-post?)))
-                             :id               id}]
+                             :locked?                      locked?
+                             :hide-if-permissions-not-met? (and hideIfPermissionsNotMet locked?)
```
Comment (flexsurfer) [inline src/status_im/subs/communities.cljs]: `:hide-if-permissions-not-met?`

**Review by flexsurfer — CHANGES_REQUESTED** [review]

File: src/status_im/subs/communities.cljs (hunk 1 above)
Comment (ibrkhalil) [reply src/status_im/subs/communities.cljs]: I'll look into the attribute's renaming flow again, But I am sure that's exactly what was destructured as.
Maybe the renaming didn't happen correctly.

Comment (ibrkhalil) [reply src/status_im/subs/communities.cljs]: Fixed here https://github.com/status-im/status-mobile/pull/19378/commits/998b8c29ed15a11a9caa9ad3d61964a0ebb8e8ef

**Review by flexsurfer — APPROVED** [review]

**Comment by pavloburykh** [comment]
@ibrkhalil thank you for the fix! Ready for merge

**Pull request merged**